src/agent/communicator/src/communicator.cpp

```diff
@@ -0,0 +1,159 @@
+#include "communicator.hpp"
+#include <boost/asio/connect.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/ssl/stream.hpp>
+#include <boost/beast/version.hpp>
+#include <iostream>
+#include <jwt-cpp/jwt.h>
+#include <nlohmann/json.hpp>
+
+#include <chrono>
+#include <iostream>
+
+using tcp = boost::asio::ip::tcp;
+using json = nlohmann::json;
+
+namespace communicator
+{
+
+    Communicator::Communicator(const std::function<std::string(std::string, std::string)> GetStringConfigValue)
+        : m_managerIp("localhost")
```

Comment:
Don't use default values for IP and port. If not present, agent should exit execution.

Reply:
Done